tests/integration/conftest.py

```diff
@@ -0,0 +1,102 @@
+import logging
+import shlex
+from pathlib import Path
+from typing import Tuple, Union
+
+import pytest
+from lightkube import AsyncClient, codecs
+from lightkube.config.kubeconfig import KubeConfig
+from lightkube.generic_resource import create_namespaced_resource
+from lightkube.resources.core_v1 import Pod
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope="module")
+async def kubeconfig(ops_test):
+    kubeconfig_path = ops_test.tmp_path / "kubeconfig"
+    retcode, stdout, stderr = await ops_test.run(
+        "juju",
+        "scp",
+        "kubernetes-control-plane/leader:/home/ubuntu/config",
+        kubeconfig_path,
+    )
+    if retcode != 0:
+        log.error(f"retcode: {retcode}")
+        log.error(f"stdout:\n{stdout.strip()}")
+        log.error(f"stderr:\n{stderr.strip()}")
+        pytest.fail("Failed to copy kubeconfig from kubernetes-control-plane")
+    assert Path(kubeconfig_path).stat().st_size, "kubeconfig file is 0 bytes"
+    yield kubeconfig_path
+
+
+@pytest.fixture(scope="module")
+async def kubernetes(kubeconfig):
+    config = KubeConfig.from_file(kubeconfig)
+    client = AsyncClient(
+        config=config.get(context_name="juju-context"),
+        trust_env=False,
+    )
+    yield client
+
+
+@pytest.fixture(scope="module")
+async def cilium_np_resource(kubernetes):
+    return create_namespaced_resource(
+        "cilium.io", "v2", "CiliumNetworkPolicy", "ciliumnetworkpolicies"
+    )
+
+
+@pytest.fixture(scope="module")
+async def hubble_test_resources(kubernetes, cilium_np_resource):
+    log.info("Creating Hubble test resources...")
+    path = Path("tests/data/hubble-test.yaml")
+    pods = []
+    for obj in codecs.load_all_yaml(path.read_text()):
+        if obj.kind == "Pod":
+            pods.append(obj.metadata.name)
+        await kubernetes.create(obj)
+
+    for pod in pods:
+        await kubernetes.wait(
+            Pod,
+            pod,
+            for_conditions=["Ready"],
+            namespace="default",
+        )
+
+    yield pods
+
+    log.info("Deleting Hubble test resources...")
+    for obj in codecs.load_all_yaml(path.read_text()):
+        await kubernetes.delete(type(obj), obj.metadata.name, namespace=obj.metadata.namespace)
+
+
+@pytest.fixture(scope="module")
+def kubectl(ops_test, kubeconfig):
+    """Supports running kubectl exec commands."""
+    KubeCtl = Union[str, Tuple[int, str, str]]
```

Comment:
wow, this is an odd type.  

it's either a `str`, or a tuple of a non-zero int and two `str`

I think this means:
it's either `stdout` or  `returncode, stdout, and stderr`

I kinda wonder if `ops_test.run` can do a better job here

All-the same, you won't have to ignore the `N806` errors if you just move 

`KubeCtl = Union[str, Tuple[int, str, str]]` outside of the fixture -- say to line 74.  But that just complicates things.  maybe just drop the type hint for the time being since it doesn't do much for you inside this fixture. 

```suggestion
```